python/lsst/ap/association/ephemerisQuery.py

```diff
@@ -0,0 +1,247 @@
+pd.options.mode.chained_assignment = 'raise'
+
+
+class EphemerisQueryConnections(PipelineTaskConnections,
+                                dimensions=("instrument",
+                                            "visit")):
+    visitInfos = connTypes.Input(
+        doc="Information defining the visit on a per detector basis.",
+        name="raw.visitInfo",
+        storageClass="VisitInfo",
+        dimensions=("instrument", "exposure", "detector"),
+        deferLoad=True,
+        multiple=True
+    )
+    ssObjects = connTypes.Output(
+        doc="Solar System objects observable in this visit retrieved from "
+            "SkyBoty",
+        name="visitSsObjects",
+        storageClass="DataFrame",
+        dimensions=("instrument", "visit"),
+    )
+
+
+class EphemerisQueryConfig(PipelineTaskConfig,
+                           pipelineConnections=EphemerisQueryConnections):
+    observerCode = pexConfig.Field(
+        dtype=str,
+        doc="IAU Minor Planet Center observer code for LSST.",
+        default='I11'
+    )
+    queryRadiusDegrees = pexConfig.Field(
+        dtype=float,
+        doc="On sky radius for Ephemeris cone search. Also limits sky "
+            "position error in ephemeris query. Defaults to the radius of "
+            "Rubin Obs FoV in degrees",
+        default=1.75)
+
+
+class EphemerisQueryTask(PipelineTask):
+    """Tasks to query the SkyBot service and retrieve the solar system objects
+    that are observable within the input visit.
+    """
+    ConfigClass = EphemerisQueryConfig
+    _DefaultName = "EphemerisQuery"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        inputs["visit"] = butlerQC.quantum.dataId["visit"]
+
+        outputs = self.run(**inputs)
+
+        butlerQC.put(outputs, outputRefs)
+
+    @pipeBase.timeMethod
+    def run(self, visitInfos, visit):
+        """Parse the information on the current visit and retrieve the
+        observable solar system objects from SkyBot.
+
+        Parameters
+        ----------
+        visitInfos : `list` of `lsst.daf.butler.DeferredDatasetHandle`
+            Set of visitInfos for raws covered by this visit/exposure. We
+            only use the first instance to retrieve the exposure boresight.
+        visit : `int`
+            Id number of the visit being run.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Results struct with components:
+
+            - ``ssObjects``: `pandas DataFrame`
+                Contains pandas DataFrame with name, RADEC position, position
+                uncertainty, and unique Id of on sky of Solar System Objects in
+                Field of View as retrieved by SkyBot.
+        """
+        # Grab the visitInfo from the raw to get the information needed on the
+        # full visit.
+        visitInfo = visitInfos[0].get(
+            datasetType=self.config.connections.visitInfos,
+            immediate=True)
```

Comment:
Unless you can point me at the place this is documented, I'm not going to change it.

Reply:
`visitInfos` is a collection of `DeferredDatasetHandle`, which only takes `component` and `parameters`, while the `*kwargs` doc says:
```
            This argument is deprecated and only exists to support legacy
            gen2 butler code during migration. It is completely ignored
            and will be removed in the future.
```

So that's why this line works at all.